backend/app/views.py

```
@@ -625,113 +623,211 @@ def index_view(request, instance=None):
-        items = request.POST.get("items")
-        instance_url = request.POST.get("url", None)
+        portal_query = Q()
+        if instance_alias:
+            portal_query |= Q(alias=instance_alias)
+        if instance_url:
+            portal_query |= Q(url=instance_url)
+
+        if not portal_query:
+            logger.error("Refresh portal called without instance alias or URL.")
+            return HttpResponse(status=200, headers={
+                "HX-Trigger-After-Settle": json.dumps({"showDangerAlert": "Portal identifier missing."})
```

Ruff (0.13.1)

`json` may be undefined, or defined from star imports

(F405)